packages/node/src/node.ts

```diff
@@ -1 +1,99 @@
-export default class Node {}
+import * as cf from "@counterfactual/cf.js";
+import machine from "@counterfactual/machine";
+import * as ethers from "ethers";
+import EventEmitter from "eventemitter3";
+import firebase from "firebase";
+
+export default class Node {
+  // @ts-ignore
+  private instructionExecutor: machine.InstructionExecutor = Object.create(
+    null
+  );
+  private eventEmitter: EventEmitter;
+
+  constructor(
+    // @ts-ignore
+    private privateKey: string,
+    // @ts-ignore
+    private provider: ethers.providers.Web3Provider,
+    // @ts-ignore
+    private firestore: firebase.firestore.Firestore,
+    // @ts-ignore
+    private networkContext: Map<string, cf.types.Address>
+  ) {
+    this.eventEmitter = new EventEmitter();
+    this.registerListeners();
+  }
+
+  /**
+   * Create a multisig for the channel with the peer specified.
+   * @param peerAddress The peer to create the multisig with.
+   */
+  async createMultisig(
+    peerAddress: cf.types.Address
+  ): Promise<cf.types.Address> {
+    // TODO:
+    return Promise.resolve(ethers.constants.AddressZero);
+  }
+
+  /**
+   * Delegates setting up a listener to the Node's EventEmitter instance.
+   * @param event
+   * @param callback
+   */
+  on(event: string, callback: (res: cf.types.Node.MethodResponse) => void) {
```

Comment:
1) Because that's where the protocol spec is. It'd be weird to have the spec be in a different package from the type definitions.
2) I see CFjs as the "common package" between other CF packages. In which case it would make to house Node protocol types as they're used by CFjs, Node and NodeProvider.

Reply:
This is a good case for the `typescript-typings` package I think. Feels like we should put types that multiple packages rely on inside there.

Reply:
I can extract the Node types to make the syntax better here, and this can be updated whenever/if the types move out.

Reply:
@cf19drofxots Do that, i'll handle the moving of the types in a later PR.